src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/UISettings.cs

```diff
@@ -0,0 +1,187 @@
+                                result &= TryGetColorValue(UISettingsRCW.UIColorType.AccentDark1, out _accentDark1);
+                                result &= TryGetColorValue(UISettingsRCW.UIColorType.AccentDark2, out _accentDark2);
+                                result &= TryGetColorValue(UISettingsRCW.UIColorType.AccentDark3, out _accentDark3);
+                                _accentColor = systemAccent;
+                            }
+                            // If result is false, hence atleast one call, use fallback values.
+                            _useFallbackColor = !result;
+                        }
+                    }
+                    catch
+                    {
+                        // We don't want to throw any exceptions here.
+                        // If we can't get any one of the color values, we will use the default accent color.
+                    }
+                }
+            }
+
+            /// <summary>
+            ///   Gets the WinRT instance of UISettings.
+            /// </summary>
+            private static object GetWinRTInstance()
+            {
+                object _winRtInstance = null;
+                try
+                {
+                    _winRtInstance = UISettingsRCW.GetUISettingsInstance();
+                }
+                catch (Exception e) when (e is TypeLoadException || e is FileNotFoundException)
+                {
+                    _winRtInstance = null;
+                }
+
+                return _winRtInstance;
+            }
+
+            #region Color Properties
+
+            internal Color AccentColor => _useFallbackColor ? _fallbackAccentColor : _accentColor;
+            internal Color AccentLight1 => _useFallbackColor ? _fallbackAccentColor : _accentLight1;
+            internal Color AccentLight2 => _useFallbackColor ? _fallbackAccentColor : _accentLight2;
+            internal Color AccentLight3 => _useFallbackColor ? _fallbackAccentColor : _accentLight3;
+            internal Color AccentDark1 => _useFallbackColor ? _fallbackAccentColor : _accentDark1;
+            internal Color AccentDark2 => _useFallbackColor ? _fallbackAccentColor : _accentDark2;
+            internal Color AccentDark3 => _useFallbackColor ? _fallbackAccentColor : _accentDark3;
+
+            #endregion
+
+            #region IDisposable
+
+            bool _disposed = false;
+
+            ~UISettings()
+            {
+                Dispose(false);
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+            private void Dispose(bool disposing)
+            {
+                if (!_disposed)
+                {
+                    if (_uisettings != null)
+                    {
+                        try
+                        {
+                            // Release the input pane here
+                            Marshal.ReleaseComObject(_uisettings);
```

Comment:
Comment is a left-over. You should make sure that if you create two instances of UISettings, disposing one will not break the other's _uisettings.

Reply:
Since, _uiSettings is not static, this won't be an issue right ?

Reply:
Not necessarily, depends on whether the native code will give you the same COM object or not.

From the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.releasecomobject):
> This risk is compounded when the COM component that is being used is a singleton, for the following reason: The CLR activates COM components by calling the COM [CoCreateInstance](https://go.microsoft.com/fwlink/?LinkID=142894) function, which returns the same interface pointer every time it is called for singleton COM components.

You are not activating the COM component yourself, but whether the WinRT gives you the same pointer or not you have to try.

Reply:
I will make a note of it for verifying and fixing it

Reply:
I verified this by disposing one instance and calling GetColorValue from other instance, and it seems to work for now. So leaving this as it is for now.